contrib/systemd/systemd.py

```diff
@@ -0,0 +1,118 @@
+import json
+import os.path
+import shutil
+import datetime
+import collections
+import pwd
+import grp
+import os
+from conf import *
+from templates import *
+
+#converts the pipe data from bots:queues kvs to
+#src_queues: bots kv, #dst_queues: count kvs
+def convert_pipedata(pipe_data):
+    sqs2bot = {}
+    dsq_count = {}
+    for bot in pipe_data:
+        src_qs =  pipe_data[bot].get('source-queue','')
+        src_qs =  [src_qs,]
+        dst_qs =  pipe_data[bot].get('destination-queues',[])
+        for q in src_qs:
+            if q in sqs2bot:
+                qb = sqs2bot[q]
+                qb.append(q)
+                sqs2bot[q] = qb
+            else:
+                sqs2bot[q] = [bot,]
+
+        for q in dst_qs:
+            if q in dsq_count:
+                qb = dsq_count[q]
+                qb += 1
+                dsq_count[q] = qb
+            else:
+                dsq_count[q] = 1
+    return (sqs2bot, dsq_count)
+
+# Return only the bots which are directly connected
+# If a bot has more than one inputs or outputs stop processing
+#
+def connected_bots(bot,rc_data,pipe_data):
```

Comment:
I know. Not as of now, I am still trying to make the directly connected units (as explained in mailing list) work as connected systemd scripts. So can we please keep these? Unfortunately i cannot commit to any timeline, atleast this week looks pretty busy.